### ISSUE TITLE
DGL crashes in libxsmm with tcmalloc

### ISSUE THREAD

**Issue by daniil-sizov** [libxsmm/libxsmm#756]
The issue doesn't reproduce if libxsmm is built with GLIBC=0

Steps to reproduce:
```bash
# install pytorch
pip3 install torch==1.13.1+cpu -f https://download.pytorch.org/whl/torch_stable.html

# install tcmalloc
sudo apt-get install libgoogle-perftools4

# install & build dgl
git clone https://github.com/dmlc/dgl.git
cd dgl
git submodule update --init --recursive
mkdir build
cd build
cmake ..
make -j32
export PYTHONPATH=<dlg path>/python:$PYTHONPATH

# reproduce
cd ../examples/pytorch/rgcn-hetero
LD_PRELOAD=/lib/x86_64-linux-gnu/libtcmalloc.so.4 python3 entity_classify.py --dataset aifb
```

Output:
```
Namespace(dataset='aifb', dropout=0, gpu=-1, l2norm=0, lr=0.01, model_path=None, n_bases=-1, n_epochs=50, n_hidden=16, n_layers=2, use_self_loop=False, validation=True)
Done loading data from cached files.
start training...
free(): invalid pointer
Aborted (core dumped)
```

**Comment by alheinecke** [comment]
I'm not sure that this libxsmm which is crashing. Can you please provide a small reproducer? (not a full training scenario).

**Comment by hfp** [comment]
Quick note: the LIBXSMM submodule used/pulled by DGL seems to be ancient (master-1.16.1-1534). At that time, we still had some malloc-magic/interceptor stuff. I suggest going for any green version of our mainline development. We eagerly work on LIBXSMM v2.0 and there will be no v1.x updates/fixes anymore.

**Comment by hfp** [comment]
Additional comment: when using GNU GCC, the CMake-flow appears to be wrong, i.e., `-lmkl_intel_thread` seems to be requested although `-lmkl_gnu_thread` would be correct. I guess this may be a handcrafted issue because MKL has a proper CMake module knowing about these differences.

**Comment by hfp** [comment]
( Reproducing this now. Potentially sending PR for updating DGL. )

**Comment by hfp** [comment]
> the LIBXSMM submodule used/pulled by DGL seems to be ancient (master-1.16.1-1534).

LIBXSMM submodule in DGL is updated meanwhile and the issue is fixed. Note, this is not by luck or because it cannot be reproduced anymore, but because of fixes since v1.17 (issue was reproduced/reported for LIBXSMM 1.16.1-1534).

**Comment by hfp** [comment]
Thank you for your report and the accurate reproducer!   

Do not hesitate to report any issue you might run into.
